### PR TITLE
Added support for extra parameters

### PR DESCRIPTION
**Pull request by leshkinski**
Adding support for the following:
- DriveLetter
- Size
- FSLabel
- README updates

Please note that DriveLetter parameter is made mandatory as it allows for easier consumption within dependent resource invocation(s) and simplifies creation of multiple partitions on same raw disk.

Module will not remove conflicting configuration at any point to stop potential data loss, but this could easily be implemented with something like a -Force switch in the future.

### CODE REVIEW

File: DSCResources/MSFT_xDisk/MSFT_xDisk.psm1 [hunk 1]
```diff
@@ -28,96 +40,167 @@ function Set-TargetResource
...
 
-        Write-Verbose -Message "Successfully initialized disk number '$($DiskNumber)'."
+
+        if ($Volume)
+        {
+            Write-Verbose -Message "Successfully initialized '$($DriveLetter)'."
+        }
+        
+        # Rebooting the node is not required
+        # $global:DSCMachineStatus = 1
```
Comment (KarolKaczmarek) [inline DSCResources/MSFT_xDisk/MSFT_xDisk.psm1]: Please remove the commented out code

File: DSCResources/MSFT_xDisk/MSFT_xDisk.psm1 [hunk 2]
```diff
@@ -28,96 +40,167 @@ function Set-TargetResource
...
+        return $false
+    }
+
+    # Drive size
+    if ($Size)
+    {
+        if ($Partition.Size -ne $Size)
         {
-            write-verbose "Disk is set Offline."
+            Write-Verbose "Drive $DriveLetter size does not match defined value"
```
Comment (KarolKaczmarek) [inline DSCResources/MSFT_xDisk/MSFT_xDisk.psm1]: It's sometimes useful to display what's expected and current value

File: DSCResources/MSFT_xDisk/MSFT_xDisk.psm1 [hunk 3]
```diff
@@ -28,96 +40,167 @@ function Set-TargetResource
...
+    }
+
+    # Volume label
+    if (-not [string]::IsNullOrEmpty($FSLabel))
+    {
+        $Label = Get-Volume -DriveLetter $DriveLetter -ErrorAction SilentlyContinue | Select-Object -ExpandProperty FileSystemLabel
+        if ($Label -ne $FSLabel)
         {
-            write-verbose "Disk is set ReadOnly."
+            Write-Verbose "Volume $DriveLetter label does not match defined value"
```
Comment (KarolKaczmarek) [inline DSCResources/MSFT_xDisk/MSFT_xDisk.psm1]: same as above..
It's useful to display what's expected and current value of label

**Comment by KarolKaczmarek** [comment]
Thanks for your contribution @leshkinski!
It looks good overall, I only had few minor comments. Please update and I'll merge this PR.

**Comment by leshkinski** [comment]
Thanks for your comments Karol! I've added the changes as per your review comments. Please let me know if you have further comments.

File: DSCResources/MSFT_xDisk/MSFT_xDisk.psm1 [hunk 4]
```diff
@@ -28,96 +40,164 @@ function Set-TargetResource
...
+        return $false
+    }
+
+    # Drive size
+    if ($Size)
+    {
+        if ($Partition.Size -ne $Size)
         {
-            write-verbose "Disk is set Offline."
+            Write-Verbose "Drive $DriveLetter size does not match expected value. Current: $Size Expected: $($Partition.Size)  "
```
Comment (KarolKaczmarek) [inline DSCResources/MSFT_xDisk/MSFT_xDisk.psm1]: Isn't it the other way: $Size expected and $Partition.Size current?

Comment (leshkinski) [inline DSCResources/MSFT_xDisk/MSFT_xDisk.psm1]: Ah, yes! Thanks

**Pull request merged**